strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/AuthorizationConfigController.java

```diff
@@ -36,109 +48,90 @@
         extends BaseArtifactController
 {
 
+    static final String SUCCESSFUL_ADD_ROLE = "The role was created successfully.";
+    static final String FAILED_ADD_ROLE = "Role cannot be saved because the submitted form contains errors!";
+
+    static final String SUCCESSFUL_DELETE_ROLE = "The role was deleted.";
+    static final String FAILED_DELETE_ROLE = "Could not delete the role.";
+
+    static final String SUCCESSFUL_ASSIGN_PRIVILEGES = "The privileges were assigned.";
+
+    static final String SUCCESSFUL_ASSIGN_ROLES = "The roles were assigned.";
+
     @Inject
-    AuthorizationConfigProvider configProvider;
+    private AuthorizationConfigProvider configProvider;
 
     @Inject
-    UserService userService;
+    private UserService userService;
 
     @Inject
-    CacheManager cacheManager;
+    private CacheManager cacheManager;
 
     @Inject
-    AnonymousAuthenticationFilter anonymousAuthenticationFilter;
+    private AnonymousAuthenticationFilter anonymousAuthenticationFilter;
 
-    private AuthorizationConfig config;
+    @Inject
+    private ConversionService conversionService;
 
-    private ResponseEntity processConfig(Consumer<AuthorizationConfig> consumer)
-    {
-        return processConfig(consumer, config -> ResponseEntity.ok()
-                                                               .build());
-    }
+    // ----------------------------------------------------------------------------------------------------------------
+    // Add role
 
-    private ResponseEntity processConfig(Consumer<AuthorizationConfig> consumer,
-                                         CustomSuccessResponseBuilder customSuccessResponseBuilder)
+    @ApiOperation(value = "Used to add new roles")
+    @ApiResponses(value = { @ApiResponse(code = 200,
+                                         message = SUCCESSFUL_ADD_ROLE),
+                            @ApiResponse(code = 400,
+                                         message = FAILED_ADD_ROLE) })
+    @PostMapping(value = "/role",
+                 consumes = MediaType.APPLICATION_JSON_VALUE,
+                 produces = { MediaType.TEXT_PLAIN_VALUE,
+                              MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity addRole(@RequestBody @Validated RoleForm roleForm,
+                                  BindingResult bindingResult,
+                                  @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
-        Optional<AuthorizationConfig> configOptional = configProvider.getConfig();
-
-        if (configOptional.isPresent())
+        if (roleForm == null)
```

Comment:
It should never happen, since 

`org.springframework.web.bind.annotation.RequestBody#required` is `default true`